pywinauto/controls/wpf_controls.py

```diff
@@ -0,0 +1,1353 @@
+    def click(self):
+        """Click the Button control by raising the ButtonBase.Click event"""
+        self.raise_event('Click')
+        # Return itself so that action can be chained
+        return self
+
+    def select(self):
+        """Select the item
+
+        Usually applied for controls like: a radio button, a tree view item
+        or a list item.
+        """
+        self.set_property('IsChecked', True)
+
+        name = self.element_info.name
+        control_type = self.element_info.control_type
+        if name and control_type:
+            self.actions.log("Selected " + control_type.lower() + ' "' + name + '"')
+
+        # Return itself so that action can be chained
+        return self
+
+    # -----------------------------------------------------------
+    def is_selected(self):
+        """Indicate that the item is selected or not.
+
+        Usually applied for controls like: a radio button, a tree view item,
+        a list item.
+        """
+        return self.get_property('IsChecked')
+
+
+class ComboBoxWrapper(wpfwrapper.WPFWrapper):
+
+    """Wrap a WPF CoboBox control"""
+
+    _control_types = ['ComboBox']
+
+    # -----------------------------------------------------------
+    def __init__(self, elem):
+        """Initialize the control"""
+        super(ComboBoxWrapper, self).__init__(elem)
+
+    # -----------------------------------------------------------
+    def expand(self):
+        self.set_property('IsDropDownOpen', True)
+        return self
+
+    # -----------------------------------------------------------
+    def collapse(self):
+        self.set_property('IsDropDownOpen', False)
+        return self
+
+    # -----------------------------------------------------------
+    def is_editable(self):
+        return self.get_property('IsEditable')
+
+    # -----------------------------------------------------------
+    def is_expanded(self):
+        """Test if the control is expanded"""
+        return self.get_property('IsDropDownOpen')
+
+    # -----------------------------------------------------------
+    def is_collapsed(self):
+        """Test if the control is collapsed"""
+        return not self.get_property('IsDropDownOpen')
+
+    # -----------------------------------------------------------
+    def texts(self):
+        """Return the text of the items in the combobox"""
+        return [child.element_info.rich_text for child in self.iter_children()]
+
+    # -----------------------------------------------------------
+    def select(self, item):
+        """
+        Select the ComboBox item
+
+        The item can be either a 0 based index of the item to select
+        or it can be the string that you want to select
+        """
+        if isinstance(item, six.integer_types):
+            self.set_property('SelectedIndex', item)
+        else:
+            index = None
+            for i, child in enumerate(self.iter_children()):
```

Comment:
`i` is not used. So enumerate is not necessary (walking children twice is also not optimal).